### ISSUE TITLE
System test MVP

### ISSUE THREAD

**Issue by prtandrup** [Energinet-DataHub/geh-charges#951]
**Synopsis:**
As the Volt team
We want to execute automated system tests (business process happy flows), prior to deploying new software updates to the actor test environment
So that we will be informed about problems that potentially span across the system as a whole (across domains) and may act upon it.

**Tech notes:**
* System tests are tests that run across domains, on the system as a whole, [source](https://github.com/Energinet-DataHub/green-energy-hub/blob/main/docs/test-principals.md)
* This test is categorized as System Test and built as a xunit test project
* Dan _(@dstsyst)_ from Mighty Ducks has already started this, by setting up a skeleton (see this [branch](https://github.com/Energinet-DataHub/geh-charges/tree/dstenroejl/system-tests)) and then another branch for pipeline work (private repo, please ask Peter _(@prtandrup)_ for location)
* All communication with the system under test is done through DataHub API Management
* We should move GreenEnergyHub.Charges.SystemTests/TestFiles/RequestChangeBillingMasterData.xml and EmbeddedResourceHelper to a new project shared by IntegrationTests and SystemTests projects (already done)

**Out of scope:**
* ~~As part of the MVP, team Volt just wants to be informed about failed system test, not guarding further deployment. A general concept needs to be discussed and agreed upon across teams. Hence deployment guarding is out of scope~~
* GUI tests

**Acceptance criteria**
- [x] **AC1 // The following system test is created for BRS-031 'Create subscription'**
Given a Grid Access Provider submits a 'Charge price list request' containing a new subscription
When the Grid Access Provider peeks for new messages
Then a confirmation message is received for the particular subscription.
(Note: later down the road, we will include a check for electricity supplier(s) receives notifications for this subscription to cover the entire business process flows)
<br>

- [x] **AC2 // Executed after deployment**
Given one or more Charges system tests are ready
When deployment has been carried out to the development environment (U)
Then the system tests are executed and failed results will prevent deployment to remaining environments

<hr>

````Below is considered a bonus target - do it if time allows - otherwise please create a new story for it````
<hr>

~~- [ ] **Bonus // The following system test is created for BRS-037 'Maintain Charge Links'**
Given a Grid Access Provider submits a new 'Charge link request' containing a new Charge Link (new charge + MP ID combination)
When the Grid Access Provider peeks for new messages
Then a Confirmation message is received for the particular charge link.~~ Moved to #1265

**Comment by prtandrup** [comment]
Blocked until MessageHub 'peek' and 'dequeue' functionality is up and running again

**Comment by PerTHenriksen** [comment]
blocked by MD token implementation.

**Comment by HenrikSommer** [comment]
The out-of-scope note regarding guarding further deployment when a test fails is obsolete.
We actually do guard further deployment already.

**Comment by prtandrup** [comment]
The new system test ran successfully as part the deployment of [PR 1241](https://github.com/Energinet-DataHub/geh-charges/pull/1241), see snippet below:

![image.png](https://images.zenhubusercontent.com/60ae42d0916bb92d8b000901/e7d3c272-4000-4286-a070-83d09c7cef7e)

**Comment by prtandrup** [comment]
### AC# 2 tested that a failing system test will guard deployment to next environment.

System tests are executed after deployment to U-001. As seen below, as failing system test will block deployment to subsequent environments, hence AC2 is fulfilled.

![image.png](https://images.zenhubusercontent.com/60ae42d0916bb92d8b000901/29f39cca-cde9-499c-a55f-15d9aec85fdb)
